### ISSUE TITLE
Missing -id_params arg results in NullPointer crash

### ISSUE THREAD

**Issue by pvstodghill** [compomics/peptide-shaker#206]
Leaving the -id_params arg off a PeptideShakerCLI invocation results in an uninformative NullPointer crash.

```
java -Djava.awt.headless=true -Xmx4g -cp /storage/ps27/work/projects/proteomic-pipeline/bin/../lib/
PeptideShaker/PeptideShaker-1.13.3.jar eu.isas.peptideshaker.cmd.PeptideShakerCLI -experiment experim
ent -sample sample -replicate 1 -identification_files work/searchruns -spectrum_files work/spectrum -
out work/peptideshaker_out.cpsx -threads 1
<CompomicsError>PeptideShaker processing failed. See the PeptideShaker log for details.</CompomicsErr
or>
java.lang.NullPointerException
        at com.compomics.util.experiment.biology.EnzymeFactory.getEnzyme(EnzymeFactory.java:68)
        at com.compomics.util.experiment.identification.parameters_cli.IdentificationParametersInputB
ean.updateIdentificationParameters(IdentificationParametersInputBean.java:1666)
        at com.compomics.util.experiment.identification.parameters_cli.IdentificationParametersInputB
ean.<init>(IdentificationParametersInputBean.java:1605)
        at eu.isas.peptideshaker.cmd.PeptideShakerCLIInputBean.<init>(PeptideShakerCLIInputBean.java:
128)
        at eu.isas.peptideshaker.cmd.PeptideShakerCLI.main(PeptideShakerCLI.java:943)
```

Please protect me from myself! :-)

Thanks.

**Comment by hbarsnes** [comment]
Hi Paul,

I will look into catching this issue and displaying a better error message. However, note that the -id_params argument is mandatory (along with the -spectrum_files argument) unless these files are included in the zipped output from SearchGUI. Which we cannot know at the point when we're validating the command line, so not straightforward to catch. :)

Best regards,
Harald

**Comment by paulstodghill** [comment]
I see the problem!

I've fixed my scripts so that the -id_params argument is used, so I hope not to run into the same problem again. If someone else does, maybe they will find this report helpful.

**Comment by carlaurangasolis** [comment]
I ran into this exact same error as well. The PeptideShaker instructions state that adding spectrum files and id parameters is not mandatory if you set the SearchGUI CLI including the command "-output_data 1" so that these PeptideShaker-required files are included in the SearchGUI output. However, this is obviously not happening, since I got the exact same error report. I re-added the id parameters AND the spectrum files to the PeptideShakerCLI and this worked like a charm. The instructions should be updated to state these are indeed required, since right now the instructions state these are not required. Cheers and be well!

**Comment by hbarsnes** [comment]
As mentioned in the online documentation (and above) the -id_params and -spectrum_files parameters *are* mandatory _unless_ these files are part of a zip file input with the identification files. See the (*) note here: https://github.com/compomics/peptide-shaker/wiki/PeptideShakerCLI#a---peptideshakercli.

However, will try to catch the exception and show something more helpful when they are missing.

Note that making the two parameters mandatory is not an option as then the single zip file input containing the spectrum files and the id parameters will no longer be valid input.

**Comment by hbarsnes** [comment]
This issue seems to have been resolved with the new PeptideShaker release (v1.14.2). If this is not the case, please let us know and we'll reopen the issue.
